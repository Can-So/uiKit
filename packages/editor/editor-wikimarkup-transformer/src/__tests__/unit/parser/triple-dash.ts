import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Triple Dash', () => {
  const testCases: Array<[string, string]> = [
    ['should replace triple dash with a special symbol', 'foo --- bar'],
    [
      'should not replace with triple dashes when sticked with parenthesis or alphanumerical',
      `
        ---foo
        bar---
        (---
        ---)
      `,
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
