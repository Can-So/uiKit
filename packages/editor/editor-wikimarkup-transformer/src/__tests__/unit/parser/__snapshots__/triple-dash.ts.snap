// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Triple Dash should not replace with triple dashes when sticked with parenthesis or alphanumerical 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "---foo",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "bar---",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "(---",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "---)",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Triple Dash should replace triple dash with a special symbol 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "foo â€” bar",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
